### PR TITLE
GSoC 2022 mid term blog - Zifeng Luo

### PR DESCRIPTION
**Pull request by luozf14**
This pull request contains the blog post for the GSoC mid-term evaluation of the project
"Automatic conversion of data stored in TTree form to RNTuple", mentored by @jalopezg-r00t .

Waiting approval of @agheata  or @hegner.

### CODE REVIEW

**Comment by luozf14** [comment]
This is my mentor @jalopezg-r00t

File: _gsocblogs/2022/blog_ROOT_ZifengLuo.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+A dictionary consists of a C++ source file, which contains the type information needed by Cling and ROOT's I/O subsystem. RNTuple supports class that has a dictionary whose persistent members are themselves types with RNTuple I/O support. Currently, my code is able to migrate data stored in TTree containing user-defined classes with compiled dictionary embedded in library ``xx.so``, i.e., to do the conversion, the corresponding library must be existing. However, TTree has a feature that when it stores a class, the dictionary is also stored automatically. Hence, the next step is to find a way to load the dictionary from TTree in case the library does not exist.
+
+Converting TTree containing branches of nested types is the most difficult part, and it is expected to be complete by the end of August. Upon completion of this, we will integrate all parts into a single program that can be run as a command line tool. It will also be used as a library for TTree-to-RNTuple conversion, hence we will be working on designing the API for the library upon the completion of the first 4 parts.
```
Comment (jalopezg-r00t) [inline _gsocblogs/2022/blog_ROOT_ZifengLuo.md:89]: Perhaps also mention integration of the separate PoCs into a reusable library (+ command line tool) in the table above? (and how that affects to the schedule).

File: _gsocblogs/2022/blog_ROOT_ZifengLuo.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+As mentioned in the beginning, TTree is the most important element of ROOT; a large amount of HEP data have been stored in TTree. Therefore, I was attracted by this TTree-To-RNtuple project because its deliverable will be the most commonly used tool among HEP scientists once ROOT v7 will be released. It would be really cool if everyone uses your tool frequently.
+
+As a physicist without any computer science background, I meet a lot of troubles when I am working on this project. However, I am very lucky that I have the nicest mentor, Dr. Javier Lopez-Gomez. No matter how simple or naive my questions are, he is always considerate and full of patience to me. For example, he explained explicitly to me how the vector and array are stored in memory by drawing sketches, while others may just say "This is basic; you should have known it".
```
Comment (jalopezg-r00t) [inline _gsocblogs/2022/blog_ROOT_ZifengLuo.md:99]: Thanks for the kind words :-).  However, I would close the blog post with some kind of conclusion.

**Review by jalopezg-r00t — APPROVED** [review]
LGTM; just a some minor corrections to be made before merging.

**Pull request merged**